js/components/cohortbuilder/components/CycleToggleInput.js

```diff
@@ -0,0 +1,31 @@
+define(['knockout', 'text!./CycleToggleInputTemplate.html'], function (ko, template) {
+
+	function CycleToggleInputViewModel(params) {
+		var self = this;
+		
+		function getOptionIndex(value, options) {
+			return (value == null) ? 0 : options.indexOf(self.options.filter((option) => option.value === value)[0]);
+		}
+		
+		self.value = params.value;
+		self.options = params.options;
+		
+		self.currentIndex = ko.pureComputed(() => getOptionIndex(self.value(), self.options));
+		self.selectedOptionText = ko.pureComputed(function() {
+			return self.options[self.currentIndex()].name;
+		});
+		
+		self.nextOption = function () {
```

Comment:
Looks too similar to `js/components/cohortdefinitionviewer/components/CycleToggleInput.js`. Why not to move common parts out / have single parametrized component?